wagtail/images/blocks.py

```diff
@@ -51,3 +57,156 @@ def htmldiff(self):
                 "image_b": self.val_b,
             },
         )
+
+
+class ImageBlock(StructBlock):
+    """
+    An usage of ImageChooserBlock with support for alt text.
+    For backward compatibility, this block overrides necessary methods to change
```

Comment:
I think we’re missing further compatibility work with `StructBlockComparison` (wagtail/admin/compare.py), as currently that code is attempting to access the block’s data as if it was a StructBlock, even though its value is returned as an image.

And we need to implement the block’s `render_basic` to render an image. Most likely something like:

```python
    def render_basic(self, value, context=None):
        return self.child_blocks["image"].render_basic(value)
```

This can all be tested by adding a block to a page, saving a draft, and then going to the "History" view to compare revisions.

Reply:
Fixed in https://github.com/wagtail/wagtail/pull/12416/commits/09b65b0070299a9077f83f1e9455798cacfa4077